leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/EndpointContextUtil.java

```diff
@@ -67,28 +67,15 @@ public static EndpointContext extractContext(Identity identity) {
             } else if (identity.isRPK()) {
                 peerIdentity = new RawPublicKeyIdentity(identity.getRawPublicKey());
             } else if (identity.isX509()) {
-                /* simplify distinguished name to CN= part */
-                peerIdentity = new X500Principal("CN=" + identity.getX509CommonName());
+                if (identity.getCertificates() != null) {
+                    peerIdentity = new X509CertPath(identity.getCertificates());
+                } else {
+                    /* simplify distinguished name to CN= part */
+                    peerIdentity = new X500Principal("CN=" + identity.getX509CommonName());
```

Comment:
I would suggest to create a separate method in `X509Util` to create the CN part.

Reply:
Done.